src/explore-education-statistics-admin/src/pages/methodology/edit-methodology/content/context/MethodologyContext.tsx

```diff
@@ -0,0 +1,181 @@
+        );
+      }
+      const matchingSection = draft.methodology[sectionKey].find(
+        section => section.id === sectionId,
+      );
+      if (!matchingSection) return draft;
+      if (Array.isArray(matchingSection.content)) {
+        matchingSection.content.push(block);
+      } else {
+        matchingSection.content = [block];
+      }
+
+      return draft;
+    }
+    case 'UPDATE_SECTION_CONTENT': {
+      const { sectionContent, meta } = action.payload;
+      const { sectionId, sectionKey } = meta;
+      if (!draft.methodology?.[sectionKey]) {
+        throw new Error(
+          `UPDATE_SECTION_CONTENT: Error - Section "${sectionKey}" could not be found.`,
+        );
+      }
+
+      const matchingSection = draft.methodology[sectionKey].find(
+        section => section.id === sectionId,
+      );
+      if (matchingSection) matchingSection.content = sectionContent;
+
+      return draft;
+    }
+    case 'ADD_CONTENT_SECTION': {
+      const { section, sectionKey } = action.payload;
+      if (draft.methodology) draft.methodology[sectionKey].push(section);
+      return draft;
+    }
+    case 'SET_CONTENT': {
+      const { content, sectionKey } = action.payload;
+      if (draft.methodology) {
+        draft.methodology[sectionKey] = content;
+      }
+      return draft;
+    }
+    case 'UPDATE_CONTENT_SECTION': {
+      const { section, meta } = action.payload;
+      const { sectionId, sectionKey } = meta;
+
+      if (draft.methodology) {
+        const sectionIndex = draft.methodology[sectionKey].findIndex(
+          accordionSection => accordionSection.id === sectionId,
+        );
+        if (sectionIndex !== -1)
+          draft.methodology[sectionKey][sectionIndex] = section;
+      }
+      return draft;
+    }
+    default: {
+      return draft;
+    }
+  }
+};
+
+function MethodologyProvider({ children }: MethodologyProviderProps) {
+  const [state, dispatch] = useImmerReducer(methodologyReducer, {
+    methodology: undefined,
+    canUpdateMethodology: false,
+  });
+  return (
+    <MethodologyStateContext.Provider value={state}>
+      <MethodologyDispatchContext.Provider value={dispatch}>
+        {children}
+      </MethodologyDispatchContext.Provider>
+    </MethodologyStateContext.Provider>
+  );
+}
+
+function useMethodologyState() {
+  const context = useContext(MethodologyStateContext);
+  if (context === undefined) {
+    throw new Error(
+      'useMethodologyState must be used within a MethodologyProvider',
+    );
+  }
+  return context;
+}
+
+function useMethodologyDispatch() {
+  const context = useContext(MethodologyDispatchContext);
+  if (context === undefined) {
+    throw new Error(
+      'useMethodologyDispatch must be used within a MethodologyProvider',
+    );
+  }
+  return context;
+}
+
+export {
+  MethodologyProvider,
+  useMethodologyState,
+  useMethodologyDispatch,
+  // eslint-disable-next-line no-undef
+  MethodologyContextState,
```

Comment:
Add `export` to the types directly instead of putting them here. Types/interfaces are not real Javascript objects so this is invalid, hence the eslint warning.